sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java

```diff
@@ -213,16 +216,46 @@ public void error(SchedulerDriver driver, String message) {
         SchedulerUtils.hardExit(SchedulerErrorCode.ERROR);
     }
 
+    protected void suppressOrRevive(PlanCoordinator planCoordinator) {
+        synchronized (suppressReviveLock) {
```

Comment:
Both of the reasons you stated are the reasons.
1. Readability
1. Avoids accidentally having methods with different purposes using `this` to lock and unintentionally conflicting.